### PR TITLE
Update Spain holidays: 2023 changes

### PR DESCRIPTION
**Pull request by delaosa**

### CODE REVIEW

File: holidays/countries/spain.py [hunk 1]
```diff
@@ -171,22 +177,36 @@ def _populate(self, year):
...
+                self._is_observed(
+                    _islamic_to_gre(year, 12, 10)[0] + rd(days=+1),
+                    "Eid al-Adha",
+                )
         elif self.subdiv == "CM":
-            if year >= 2022:
+            if year == 2022:
                 self._is_observed(date(year, JUN, 16), "Corpus Christi")
+            elif year == 2023:
+                self._is_observed(date(year, JUN, 8), "Corpus Christi")
```
Comment (KJhellico) [inline holidays/countries/spain.py:193]:
```suggestion
            if year >= 2022:
                self._is_observed(easter_date + rd(days=+60), "Corpus Christi")
```

File: holidays/countries/spain.py [hunk 2]
```diff
@@ -54,6 +54,21 @@ def _populate(self, year):
...
+                "AN",
+                "AR",
+                "AS",
+                "CL",
+                "MC",
+            }
+        ):
+            self._is_observed(date(year, MAR, 19), "Año nuevo")
+        if year != 2023:
+            self._is_observed(date(year, MAR, 19), "Año nuevo")
```
Comment (KJhellico) [inline holidays/countries/spain.py:71]: "Año nuevo"?

File: holidays/countries/spain.py [hunk 3]
```diff
@@ -53,70 +53,77 @@ def _is_observed(self, date_holiday, name_holiday):
...
 
-        self._is_observed(date(year, JAN, 1), "Año nuevo")
+        if year == 2023 and self.subdiv in {
+            "AN",
+            "AR",
+            "AS",
+            "CL",
+            "MC",
+        }:
+            self._is_observed(date(year, JAN, 1), "Año nuevo")
```
Comment (KJhellico) [inline holidays/countries/spain.py:63]: In rest of regions there was no New Year holiday at all?

Comment (delaosa) [reply holidays/countries/spain.py:63]: Since 1th of January was Sunday, just some regions observed this festivity the day after.

Comment (delaosa) [reply holidays/countries/spain.py:63]: For reference:

https://administracion.gob.es/pag_Home/dam/jcr:a31ce637-a7ba-49e1-9f41-e7a86d8ec867/Calendario_Laboral2023.pdf

**Comment by delaosa** [comment]
it's my understanding that when a holiday happens on Sunday it's not added
to the list of holidays for that year.

File: holidays/countries/spain.py (hunk 3 above)
Comment (delaosa) [reply holidays/countries/spain.py:63]: it's my understanding that when a holiday happens on Sunday it's not added
to the list of holidays for that year.


For reference:

https://administracion.gob.es/pag_Home/dam/jcr:a31ce637-a7ba-49e1-9f41-e7a86d8ec867/Calendario_Laboral2023.pdf

File: holidays/countries/spain.py [hunk 4]
```diff
@@ -137,7 +150,11 @@ def _populate(self, year):
             "MC",
         }:
             self._is_observed(date(year, MAY, 1), "Día del Trabajador")
-        if self.subdiv in {"CT", "GA", "VC"}:
+        if year != 2022:
+            self._is_observed(date(year, MAY, 1), "Día del Trabajador")
+        if year == 2023 and self.subdiv in {"CT", "VC"}:
+            self._is_observed(date(year, JUN, 24), "San Juan")
+        if year != 2023 and self.subdiv in {"CT", "GA", "VC"}:
             self._is_observed(date(year, JUN, 24), "San Juan")
```
Comment (KJhellico) [inline holidays/countries/spain.py:158]:
```suggestion
        if self.subdiv in {"CT", "VC"} or (
            year != 2023 and self.subdiv == "GA"
        ):
            self._is_observed(date(year, JUN, 24), "San Juan")
```

File: holidays/countries/spain.py (hunk 3 above)
Comment (arkid15r) [reply holidays/countries/spain.py:63]: I believe it's worth adding `https://administracion.gob.es/pag_Home/atencionCiudadana/calendarios.html` to Spain docstring as a reference.

File: holidays/countries/spain.py [hunk 5]
```diff
@@ -125,10 +132,16 @@ def _populate(self, year):
             "VC",
         }:
             self[easter(year) + rd(days=+1)] = "Lunes de Pascua"
-
-        if 2022 != year:
-            self._is_observed(date(year, MAY, 1), "Día del Trabajador")
-        elif 2022 == year and self.subdiv in {
+        elif year >= 2022 and self.subdiv in {
+            "RI",
```
Comment (arkid15r) [inline holidays/countries/spain.py:136]: Could you sort these for consistency with rest of the code?

File: holidays/countries/spain.py [hunk 6]
```diff
@@ -210,31 +241,29 @@ def _populate(self, year):
...
                     _islamic_to_gre(year, 12, 10)[0] + rd(days=+2),
-                    "Aid Al-Adha",
-                )
-            else:
-                self._is_observed(
-                    _islamic_to_gre(year, 10, 1)[0], "Aid Al-Fitr"
+                    "Eid al-Adha",
                 )
+            elif year == 2023:
+                self._is_observed(_islamic_to_gre(year, 10, 1)[0], "Eid Fitr")
```
Comment (arkid15r) [inline holidays/countries/spain.py:251]: Optional: Probably you know these holiday names better. My approach to naming  would be either "Eid Fitr, Aid Al Adha" (as per the reference you've provided) or "Eid al-Fitr, Eid al-Adha" (as per the Wikipedia).

File: test/countries/test_spain.py [hunk 7]
```diff
@@ -265,6 +206,115 @@ def test_variable_days_in_2022(self):
                     % (date(2022, *fest_date), prov, ", ".join(fest_provs)),
                 )
 
+    def test_variable_days_in_2023(self):
+        province_days = {
+            (1, 2): [
```
Comment (arkid15r) [inline test/countries/test_spain.py:211]: Optional: if you're in a mood to go an extra mile it'd be great to have months replaced with their constants from `constants.py` (e.g., (1, 2) -> (JAN, 2), (2, 21) -> (FEB, 21)

File: test/countries/test_spain.py [hunk 8]
```diff
@@ -265,6 +206,115 @@ def test_variable_days_in_2022(self):
...
+                "AN",
+                "AR",
+                "AS",
+                "CL",
+                "MC",
+            ],
+            (1, 6): [
```
Comment (arkid15r) [inline test/countries/test_spain.py:218]: I believe it's okay to use a set here too (and below)

**Review by arkid15r — CHANGES_REQUESTED** [review]
Hi @delaosa, thank you for fixing such a lot of discrepancies for Spain!

I added a couple of minor issues to address. Please let me know if you have any questions.

File: test/countries/test_spain.py (hunk 7 above)
Comment (delaosa) [reply test/countries/test_spain.py]: Done

File: test/countries/test_spain.py (hunk 8 above)
Comment (delaosa) [reply test/countries/test_spain.py]: Done

File: holidays/countries/spain.py (hunk 6 above)
Comment (delaosa) [reply holidays/countries/spain.py]: Done

File: holidays/countries/spain.py (hunk 5 above)
Comment (delaosa) [reply holidays/countries/spain.py:137]: Done

**Comment by arkid15r** [comment]
Merged into beta!

@delaosa thanks for fixing this. @KJhellico, thanks for the reviews!

**Pull request merged**